pkgs/development/beam-modules/build-mix.nix

```diff
@@ -1,100 +1,76 @@
-{ stdenv, writeText, elixir, erlang, hex, lib }:
+{ stdenv, lib, elixir, erlang, hex, rebar, rebar3, fetchMixDeps, makeWrapper }:
 
 { name
 , version
 , src
-, setupHook ? null
-, buildInputs ? []
-, beamDeps ? []
-, postPatch ? ""
-, compilePorts ? false
-, installPhase ? null
-, buildPhase ? null
-, configurePhase ? null
-, meta ? {}
+, nativeBuildInputs ? [ ]
+, meta ? { }
+, buildEnvVars ? { }
 , enableDebugInfo ? false
-, ... }@attrs:
-
-with lib;
-
+, depsSha256
+, mixEnv ? "prod"
+, ...
+}@attrs:
 let
+  mixDeps = fetchMixDeps {
+    inherit src name mixEnv version buildEnvVars;
+    sha256 = depsSha256;
+  };
 
-  debugInfoFlag = lib.optionalString (enableDebugInfo || elixir.debugInfo) "--debug-info";
-
-  shell = drv: stdenv.mkDerivation {
-          name = "interactive-shell-${drv.name}";
-          buildInputs = [ drv ];
-    };
-
-  bootstrapper = ./mix-bootstrap;
-
-  pkg = self: stdenv.mkDerivation ( attrs // {
-    name = "${name}-${version}";
-    inherit version;
-
-    dontStrip = true;
-
-    inherit src;
+  overridable = builtins.intersectAttrs { preConfigure = ""; postConfigure = ""; } attrs;
```

Comment:
I don't really like what I am doing here, if anybody has a better idea I'm open.
Basically, I just want to allow some attributes to be overriden.

Reply:
I think the new version with `removeAttrs attrs` is good, this is what I think most builderst do, e.g. the [Rust](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/default.nix#L124) builder and the [Go](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/go-packages/generic/default.nix#L74) builder. I think this is even enough for setting build time env variables, so for me, we don't need a `buildEnvVars` attribute with the new version.